x-pack/plugins/fleet/server/services/epm/packages/update.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract } from 'kibana/server';
+import type { TypeOf } from '@kbn/config-schema';
+
+import { PACKAGES_SAVED_OBJECT_TYPE } from '../../../constants';
+import type { Installation, UpdatePackageRequestSchema } from '../../../types';
+import { IngestManagerError } from '../../../errors';
+
+import { getInstallationObject, getPackageInfo } from './get';
+
+export async function updatePackage(
+  options: {
+    savedObjectsClient: SavedObjectsClientContract;
+    pkgName: string;
+    keepPoliciesUpToDate?: boolean;
+  } & TypeOf<typeof UpdatePackageRequestSchema.body>
+) {
+  const { savedObjectsClient, pkgName, keepPoliciesUpToDate } = options;
+  const installedPackage = await getInstallationObject({ savedObjectsClient, pkgName });
+
+  if (!installedPackage) {
+    throw new IngestManagerError(`package ${pkgName} is not installed`);
```

Comment:
IngestManagerError could be renamed to FleetError, if not too much trouble

Reply:
Certainly a fair point, but I'm not sure that should be in scope here. I created https://github.com/elastic/kibana/issues/113801 to serve as a running list of renames/improvements we can make to clean up references to Fleet's legacy naming.